src/js.cookie.js

```diff
@@ -66,13 +66,17 @@
 				key = key.replace(/%(23|24|26|2B|5E|60|7C)/g, decodeURIComponent);
 				key = key.replace(/[\(\)]/g, escape);
 
-				return (document.cookie = [
-					key, '=', value,
-					attributes.expires && '; expires=' + attributes.expires.toUTCString(), // use expires attribute, max-age is not supported by IE
-					attributes.path    && '; path=' + attributes.path,
-					attributes.domain  && '; domain=' + attributes.domain,
-					attributes.secure ? '; secure' : ''
-				].join(''));
+				if (typeof document !== 'undefined') {
```

Comment:
We can do something like `var document = {}` in this scope, so that it will be noop for read and write operations. In this case there will be no need to use `typeof`and `!==` and `undefined`, which are unminifiable tokens.